src/libraries/Microsoft.PowerFx.Interpreter/PatchFunction.cs

```diff
@@ -0,0 +1,187 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.DLP;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Texl.Builtins;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    /*
+         * @@@JYL@@@
+         * Verify before pushing it.
+         * => verify that Patch() fails in a non-behavior context.
+         * => Type checking - what if we pass the wrong type to patch (too few fields, too many fields, not a record, etc).
+         * => PowerApps lets pass multiple records - this is determined by CheckInvocation, which we copied from PA.
+         * => Will need to confirm corner case behavior, ie, what if we pass errors, etc. (We can probably do that in a followup PR, just to unblock usage now)
+         * => Trying to patch on an Immutable object. We should get an error.
+         */
+
+    internal abstract class PatchAndValidateRecordFunctionBase : BuiltinFunction
+    {
+        public PatchAndValidateRecordFunctionBase(DPath theNamespace, string name, TexlStrings.StringGetter description, FunctionCategories fc, DType returnType, BigInteger maskLambdas, int arityMin, int arityMax, params DType[] paramTypes)
+           : base(theNamespace, name, /*localeSpecificName*/string.Empty, description, fc, returnType, maskLambdas, arityMin, arityMax, paramTypes)
+        {
+        }
+
+        public PatchAndValidateRecordFunctionBase(string name, TexlStrings.StringGetter description, FunctionCategories fc, DType returnType, BigInteger maskLambdas, int arityMin, int arityMax, params DType[] paramTypes)
+            : this(DPath.Root, name, description, fc, returnType, maskLambdas, arityMin, arityMax, paramTypes)
+        {
+        }
+
+        public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(binding);
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+
+            var isValid = CheckInvocation(args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            return isValid;
+        }
+    }
+
+    internal class PatchRecordFunction : PatchAndValidateRecordFunctionBase, IAsyncTexlFunction
```

Comment:
>PatchRecordFunction [](http://example.com/codeflow?start=19&length=19)

Can you add a comment for which Patch() overload this is.

Reply:
Also added to PatchFunction function